### DISCUSSION TITLE
clone and detach

### DISCUSSION THREAD

**Discussion by wangjiawen2013** [tracel-ai/burn#2629] [Q&A] [answered]
Hi,
It is said that "We don't need to be worried about memory overhead because with cloning, the tensor's buffer isn't copied, and only a reference to it is increased." in burn's book, which is different from pytorch. In pytorch, `clone` will copy the tensor's buffer.

In pytorch, `detach` will share the buffer with the original tensor. what is it in burn ? Does `detach` share or copy the tensor buffer in burn ?

There are no detailed info on `clone` and `detach` in burn's book, so we want to know are they the same between pytorch and burn.

**Comment by laggui** [comment, marked as answer]
`tensor.detach` is used to detach the tensor from the current autodiff graph and creates a new leaf tensor.

The tensor is not copied.
